### ISSUE TITLE
Padauk PSF154 for 7 segment displays

### ISSUE THREAD

**Issue by svanimisetti** [brainsmoke/charliepdk#1]
This more of a question than issue ( as I could not find "Discussions" activated for this repo ).

For a while now, I have been looking at a solution to drive ~900 LEDs setup as 90 digits (7-segment number + 3 LEDs custom LEDs). I saw a great article from [Sean Hodgins](https://www.hackster.io/seanhodgins/addressable-7-segment-displays-26bbfb), but soon realized that the WS8211 is not cost-effective and more importantly is not readily available in large numbers. The project will require 360 WS8211s and the overall cost will run in a few 100$. Not feasible!

I came across an article on [Tim's blog](https://cpldcpu.wordpress.com/2020/04/05/addressable-7-segment-display/), that uses the Padauk PSF154 as an addressable microcontroller to drive LEDs. I also stumbled upon an article on [Adafruit ](https://blog.adafruit.com/2021/01/20/a-charlieplexed-led-array-driven-using-a-padauk-pfs154-microcontroller/) that seemed to use the same microcontroller - but for a different application.

I wanted to know if Adafruit has libraires/articles that support this uC. I could not find this uC listed on their product catalog - so was curious how you did your project. Also, I was not aware if these uCs have active support within the Arduino IDE/ecosystem.

**Comment by brainsmoke** [comment]
Getting ~300 ws2811 chips shouldn't be too difficult, but I would use a matrix display setup.

There's a pretty standard 16 pin LED driver chip design (I used Chipone ICN2038, but there are more alternatives) that acts like a shift register and sinks constant current (selectable using a single resistor.)

Those are usually paired with a combined multiplexer & high side mosfet (like Chipone ICN2012). Combining 8 driver chips and 1 multiplexer would allow you to drive 8*16*8 = 1024 leds. You can always choose to use more driver chips to give each LED more time to shine.

You can drive this setup using standard led matrix drivers for the esp32, or an FPGA.

If you use three or 6 data lines, you may even be able to just plug in a 'HUB75 led matrix' driver
https://www.adafruit.com/product/420

https://github.com/brainsmoke/hexboard I've used a setup with 8 chips, 4x multiplexed (so 512 LEDs) driven by an STM32

**Comment by svanimisetti** [comment]
Thank @brainsmoke. Unfortunately, I am inexperienced and still trying to learn and understand! :)

I had looked at matrix display setup, however, the LEDs that I am setting up are a part of linear display, which extends ~2m in length and has around 90 individual modules with 10leds per module. I initially thought using multiplexers, but after researching the internet I figured SPI interface over long distances will not work properly. I was told the 1-wire implementation on WS2812b or WS2811 can work really long distances provided power is added every few meters.

I am not an expert - but thanks for the direction - I will definitely look at ICN2012 and similar uCs for displays.
